pkg/event_handler/event_handler.go

```diff
@@ -0,0 +1,33 @@
+package event_handler
+
+import (
+	"fmt"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog/v2"
+)
+
+type IListener interface {
+	Run(resync func()) error
+}
+
+type IStoppableRsync interface {
+	Stop()
+}
+
+func StartEventHandler(eventListener IListener, controllerHandlerFactory func() (IStoppableRsync, error)) error {
```

Comment:
I suggest to call it something like `initControllerHandler`. Factory is confusing as it's not an implementation of factory pattern